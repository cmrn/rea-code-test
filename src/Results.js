import React from 'react';
import './Results.css';

const Results = (props) => (
  <div {...props}>
    <pre>Results go here</pre>
  </div>
);

export default Results;
