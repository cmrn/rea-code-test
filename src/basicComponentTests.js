import React from 'react';
import { shallow } from 'enzyme';

export default function basicComponentTests(component) {
  it('allows a className to be set', () => {
    const wrapper = shallow(React.cloneElement(component, { className: "classy" }));
    expect(wrapper.props().className).toEqual(expect.stringContaining('classy'));
  });

  it('passes through other props', () => {
    const wrapper = shallow(React.cloneElement(component, { foo: "bar" }));
    expect(wrapper.props().foo).toEqual('bar');
  });
}
