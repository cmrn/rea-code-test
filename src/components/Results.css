.Results-hover {
  color: white;
  font-size: 25px;
  text-align: center;
}

.Results-hover-icon {
  width: 75px;
  height: 75px;
}