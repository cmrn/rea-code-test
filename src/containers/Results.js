import React from 'react';
import PropertyList from '../components/PropertyList';
import { getResults } from '../api';


class Results extends React.Component {
  constructor(props) {
    super(props);
    this.state = { results: [] };
  }

  async componentDidMount() {
    const results = await getResults();
    this.setState({ results });
  }

  render() {
    return (
      <PropertyList 
        {...this.props} 
        properties={this.state.results} 
        title='Results' 
        hoverContent={<div>Save</div>}
        onClick={(id) => console.log(id)}
      />
    );
  }
};

export default Results;
