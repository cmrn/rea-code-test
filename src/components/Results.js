import React from 'react';
import PropTypes from 'prop-types';
import './Results.css';

const Results = ({results, ...props}) => (
  <div {...props}>
    <code>{JSON.stringify(results)}</code>
  </div>
);

Results.propTypes = {
  results: PropTypes.array.isRequired,
};

export default Results;
