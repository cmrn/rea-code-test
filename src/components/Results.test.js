import React from 'react';
import basicComponentTests from '../basicComponentTests';
import { shallow } from 'enzyme';
import Results from './Results';

const dummyProps = {
  propertyIds: []
};

basicComponentTests(<Results {...dummyProps} />);

it('sets title and hover content as expected', () => {
  const wrapper = shallow(<Results {...dummyProps} />);
  expect(wrapper).toMatchSnapshot();
});
