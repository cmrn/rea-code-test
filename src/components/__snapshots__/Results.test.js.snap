// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`with no results renders no properties 1`] = `
<div>
  <code>
    []
  </code>
</div>
`;

exports[`with some results renders the properties in the results 1`] = `
<div>
  <code>
    [{"price":"$726,500","agency":{"brandingColors":{"primary":"#ffe512"},"logo":"http://i1.au.reastatic.net/agencylogo/XRWXMT/12/20120927204448.gif"},"id":"1","mainImage":"http://i2.au.reastatic.net/640x480/20bfc8668a30e8cabf045a1cd54814a9042fc715a8be683ba196898333d68cec/main.jpg"},{"price":"$560,520","agency":{"brandingColors":{"primary":"#fcfa3b"},"logo":"http://i4.au.reastatic.net/agencylogo/BFERIC/12/20150619122858.gif"},"id":"2","mainImage":"http://i1.au.reastatic.net/640x480/88586227f9176f602d5c19cf06261108dbb29f03e30d1c4ce9fc2b51fb1e4bd6/main.jpg"}]
  </code>
</div>
`;
